### PR TITLE
add script to archive logs from all running docker containers

### PR DESCRIPTION
**Pull request by mrjones-plip**
# Description

There was a [discussion on Slack](https://medic.slack.com/archives/C02F2TAME90/p1674141851263569) recently about having a script we could encourage folks to run who were having troubles running CHT 4.x's docker containers. The discussion was in response to [a forum post](https://forum.communityhealthtoolkit.org/t/error-502-when-running-local-cht-instance/2414).  We thought the script would gather up all the logs and create a .tar.gz file that could easily be posted.  This PR adds a bash script to do just that. 

To test:
* create a CHT 4.x instance using[ docker helper](https://docs.communityhealthtoolkit.org/apps/guides/hosting/4.x/app-developer/#cht-docker-helper-for-4x)
* clone the `cht-core` repo, and check out the `add-script-to-zip-docker-logs` branch
* Change directories into the `cht-core` repo: `cd cht-core`
* run the script `./scripts/compress_and_archive_docker_logs.sh`
* check the `support_logs` directory and ensure there's a zip file in there: `ls ~/.medic/support_logs/`
* uncompressing the tarball should yield one log file per container with the container name as the file name. Also a file for `docker ps` and `docker stats` should be saved as `docker_ps.log` and `docker_stats.log`respectively.
* check that the `*log` files have correct contents

# Code review checklist

- [X] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [X] Tested: Unit and/or e2e where appropriate

# Compose URLs

If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:add-script-to-zip-docker-logs/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:add-script-to-zip-docker-logs/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:add-script-to-zip-docker-logs/docker-compose/cht-couchdb-clustered.yml)
 
# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: scripts/zip_docker_logs.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#/bin/bash
+
+##############################################3
+# This script gathers logs for ALL containers
+# running and outputs them into a zip file in
+# your home directory here:
+#
+#       ~/.medic/cht-docker/support_logs
```
Comment (dianabarsan) [inline scripts/zip_docker_logs.sh:8]:
```suggestion
# ~/.cht/docker/support_logs
```

Maybe we'd want to keep "medic" out of it.

File: scripts/zip_docker_logs.sh [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#/bin/bash
+
+##############################################3
+# This script gathers logs for ALL containers
+# running and outputs them into a zip file in
+# your home directory here:
+#
```
Comment (dianabarsan) [inline scripts/zip_docker_logs.sh:7]:
```suggestion
set -e
```

File: scripts/zip_docker_logs.sh [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+mkdir -p "$log_directory"
+mkdir -p "$tmp"
+rm /tmp/cht-docker-log-tmp/*
+
+docker stats --no-stream 1> ${tmp}/docker_stats.log
+docker ps> ${tmp}/docker_ps.log
+docker ps  --format '{{ .Names }}' | xargs -I % sh -c "docker logs %  > ${tmp}/%.log 2>&1"
+
+cd /tmp/cht-docker-log-tmp
+zip --quiet ${log_archive} *
```
Comment (dianabarsan) [inline scripts/zip_docker_logs.sh:34]: This assumes everyone has `zip` installed. It might not be the case?

File: scripts/zip_docker_logs.sh [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+rm /tmp/cht-docker-log-tmp/*
+
+echo "\nDone!\n\nZip file here:\n\n  ${log_archive}\n"
```
Comment (dianabarsan) [inline scripts/zip_docker_logs.sh:38]: I'm seeing `\n` in all of these echoes. 

image

**Comment by dianabarsan** [comment]
This is awesome @mrjones-plip ! Thanks for taking the time to add this.

File: scripts/zip_docker_logs.sh (hunk 1 above)
Comment (mrjones-plip) [reply scripts/zip_docker_logs.sh:8]: Ah - yeah - I can see that.  I was leveraging the existing [directory CHT Docker Helper creates](https://github.com/medic/cht-core/blob/master/scripts/docker-helper-4.x/cht-docker-compose.sh#L68).  You think it's worth keeping these the same or diverging them?

File: scripts/zip_docker_logs.sh (hunk 3 above)
Comment (mrjones-plip) [reply scripts/zip_docker_logs.sh]: Fair!  I swapped it out for `tar` in c07e5882b

**Comment by mrjones-plip** [comment]
Thanks @dianabarsan! light push back on the directory structure and back to you for review!

File: scripts/zip_docker_logs.sh (hunk 1 above)
Comment (dianabarsan) [reply scripts/zip_docker_logs.sh]: I think keeping the same makes sense, though I do think we should eventually remove the "medic" out of it. Do you see a bunch of "Canonical" folders in your Ubuntu install?

**Review by dianabarsan — APPROVED** [review]
Very nice!

**Comment by dianabarsan** [comment]
I suppose you want this merged immediately?

**Comment by mrjones-plip** [comment]
> I suppose you want this merged immediately?

I'll whack CI to re-run failed jobs once, and if that fails, yeah, just force merge.

**Pull request merged**